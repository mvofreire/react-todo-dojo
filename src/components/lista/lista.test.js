import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";
import { Lista } from ".";

describe("Testando Component <Lista /> e <Item />", () => {
  it("Renderiza Lista sem erros", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Lista />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Valida se renderiza todos os Filhos", () => {
    const wrapper = shallow(<Lista>Teste de children</Lista>);
    expect(wrapper.text()).toContain("Teste de children");
  });

  it("Item Component to match snapshot", () => {
    const wrapper = create(<Lista />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
