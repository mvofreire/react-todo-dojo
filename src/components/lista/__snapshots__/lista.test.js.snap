// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando Component <Lista /> e <Item /> Item Component to match snapshot 1`] = `
<div
  className="lista"
/>
`;
